### PR TITLE
Added zlib compression to XMLStorageManager

### PR DESCRIPTION
**Pull request by regislebrun**
Using even the weakest compression leads to a 6 times smaller file!

### CODE REVIEW

File: lib/src/Base/Common/openturns/Study.hxx [hunk 1]
```diff
@@ -67,6 +67,15 @@ public:
...
 
+#if defined OPENTURNS_HAVE_LIBXML2
+
+  /**
+   * Parameter constructor
+   */
+  Study(const String & fileName);
+
+#endif /* OPENTURNS_HAVE_LIBXML2 */
+
```
Comment (jschueller) [inline lib/src/Base/Common/openturns/Study.hxx]: OPENTURNS_HAVE_XXX macros are unavailable when compiling outside the library, hence I suggest to not use it in the header ; leave the constructor unconditionally in the header and use the macro inside the constructor code.

File: lib/test/t_Study_saveload.cxx [hunk 2]
```diff
@@ -59,6 +59,17 @@ int main(int argc, char *argv[])
...
+    {
+      Study study(fileName);
+    }
+
+    // Create a Study Object by giving its storage manager with a nonzero compression level
+    {
+      Study study;
+      study.setStorageManager(XMLStorageManager(fileName, 5));
+    }
+
```
Comment (regislebrun) [inline lib/test/t_Study_saveload.cxx]: @dbarbier It is just to check that it is possible to create an XMLStorageManager using a given compression level and to set it into a Study instance.

File: lib/src/Base/Common/openturns/Study.hxx [hunk 3]
```diff
@@ -67,11 +67,16 @@ public:
...
+   * Parameter constructor
+   */
+  Study(const String & fileName);
+
   /** @copydoc Object::__repr__() const */
   virtual String __repr__() const;
 
   /** @copydoc Object::__str__() const */
-  virtual String __str__(const String & offset = "") const;
+  virtual String __str__(const FileName & offset = "") const;
```
Comment (regislebrun) [inline lib/src/Base/Common/openturns/Study.hxx]: @dbarbier It has been fixed since your review.

File: python/src/XMLStorageManager_doc.i.in [hunk 4]
```diff
@@ -24,6 +28,8 @@ Examples
 >>> # Create a study:
 >>> myStudy = ot.Study()
 >>> myStudy.setStorageManager(ot.XMLStorageManager('myStudy.xml'))
+>>> myStudy2 = ot.Study()
+>>> myStudy2.setStorageManager(ot.XMLStorageManager('myStudy2.xml', 5))
```
Comment (regislebrun) [inline python/src/XMLStorageManager_doc.i.in]: @dbarbier You are right. Changed.

**Comment by jschueller** [comment]
This can be rebased now, as #489 fixed the Travis build.

File: lib/test/t_Study_saveload.cxx (hunk 2 above)
Comment (dbarbier) [inline lib/test/t_Study_saveload.cxx]: You only check that the constructor exists, IMO this is useless.

**Comment by regislebrun** [comment]
@dbarbier You are right, I updated the tests.

File: python/src/Study_doc.i.in [hunk 5]
```diff
@@ -1,6 +1,18 @@
...
 "Study.
 
+Available constructors:
+    Study()
+
+    Study(*fileName*)
+
+Parameters
+----------
+fileName : string
```
Comment (jschueller) [inline python/src/Study_doc.i.in]: str

File: python/src/XMLStorageManager_doc.i.in [hunk 6]
```diff
@@ -4,10 +4,14 @@
...
     XMLStorageManager(*filename*)
 
+    XMLStorageManager(*filename, compressionLevel*)
+
 Parameters
 ----------
 fileName : str
     Name of the file containing the study.
+compressionLevel : int
+    Compression level of the resulting XML file, in the range 0,...,9 (0=no compression, 9=best compression). The compression is done only if the XML lib has been compiled with zlib support. If not given, the compression level is equal to :class:`~openturns.ResourceMap.GetAsUnsignedInteger('XMLStorageManager-DefaultCompressionLevel')` with a default value of zero.
```
Comment (jschueller) [inline python/src/XMLStorageManager_doc.i.in]: line too long,
you can only mention the resourcemap key name, not :class:`~openturns.ResourceMap.GetAsUnsignedInteger('XMLStorageManager-DefaultCompressionLevel')`

**Pull request merged**